agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilderTest.java

```diff
@@ -135,4 +135,69 @@ void testRpConfigurationOverlayWithEnvVarAndSysPropPopulated() throws Exception
               assertThat(config.sampling.percentage).isEqualTo(testSamplingPercentage);
             });
   }
+
+  private static final String CONNECTION_STRING =
+      "InstrumentationKey=00000000-0000-0000-0000-000000000000;IngestionEndpoint=https://fake-ingestion-endpoint";
+
+  // "${file:file-look-up-connection-string.txt}"
+  @Test
+  void testOverlayWithEnvVarWithGoodFileStringLookupFormat() throws Exception {
+    File connectionStringFile =
+        new File(
+            getClass()
+                .getClassLoader()
+                .getResource("file-look-up-connection-string.txt")
+                .getPath());
+    Configuration configuration = new Configuration();
+    configuration.connectionString = "${file:" + connectionStringFile.getPath() + "}";
+    ConfigurationBuilder.overlayFromEnv(configuration);
+    assertThat(configuration.connectionString).isEqualTo(CONNECTION_STRING);
+  }
+
+  @Test
+  void testOverlayWithEnvVarWithBadFileStringLookupFormat() throws Exception {
+    File connectionStringFile =
+        new File(
+            getClass()
+                .getClassLoader()
+                .getResource("file-look-up-connection-string.txt")
+                .getPath());
+    Configuration configuration = new Configuration();
+    configuration.connectionString = "${file:" + connectionStringFile.getPath();
+    ConfigurationBuilder.overlayFromEnv(configuration);
+    assertThat(configuration.connectionString).isEqualTo(configuration.connectionString);
+
+    configuration.connectionString = "file:" + connectionStringFile.getPath() + "}";
+    ConfigurationBuilder.overlayFromEnv(configuration);
+    assertThat(configuration.connectionString).isEqualTo(configuration.connectionString);
+
+    configuration.connectionString = "file:" + connectionStringFile.getPath();
+    ConfigurationBuilder.overlayFromEnv(configuration);
+    assertThat(configuration.connectionString).isEqualTo(configuration.connectionString);
+
+    configuration.connectionString = CONNECTION_STRING;
+    ConfigurationBuilder.overlayFromEnv(configuration);
+    assertThat(configuration.connectionString).isEqualTo(configuration.connectionString);
+  }
+
+  @Test
+  void testConnectionStringEnvVarHasHigherPrecedenceOverFileLookup() throws Exception {
+    File connectionStringFile =
+        new File(
+            getClass()
+                .getClassLoader()
+                .getResource("file-look-up-connection-string.txt")
+                .getPath());
```

Comment:
consolidate on one way to test from file? e.g. either resource file like here or temp file like in other test?

Reply:
sure, i can use temp file like FileStringLookup.. too many resource files already.